modules/light-clients/06-solomachine/client_state.go

```diff
@@ -112,17 +112,17 @@ func (cs *ClientState) VerifyMembership(
 	delayTimePeriod uint64,
 	delayBlockPeriod uint64,
 	proof []byte,
-	path []byte,
+	path exported.Path,
 	value []byte,
 ) error {
 	publicKey, sigData, timestamp, sequence, err := produceVerificationArgs(cdc, cs, height, proof)
 	if err != nil {
 		return err
 	}
 
-	var merklePath commitmenttypes.MerklePath
-	if err := cdc.Unmarshal(path, &merklePath); err != nil {
-		return sdkerrors.Wrap(commitmenttypes.ErrInvalidProof, "failed to unmarshal path into ICS 23 commitment merkle path")
+	merklePath, ok := path.(commitmenttypes.MerklePath)
+	if !ok {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "expected %T, got %T", commitmenttypes.MerklePath{}, path)
 	}
```

Comment:
Code coverage both here and in `07-tendermint` will likely be reduced. I can cover the error case here by adding a mock path type to the tests like so:

```go
type mockPath struct{}

func (mockPath) String() string {
	return ""
}

func (mockPath) Empty() bool {
	return false
}
```

Reply:
fwiw I don't feel super strongly about covering this case...

Reply:
No preferences, probably doesn't hurt to add. Hopefully one day it can be removed. I'm fine if it is skipped as well